### PR TITLE
Review of alt text content and translatability

### PR DESCRIPTION
**Pull request by microbit-robert**
- Alt text for purely decorative images has been removed
- All img tags have an alt attribute (empty string if no text is applicable)
- All alt text is now translatable
- Alt text has been updated using https://paper.dropbox.com/doc/ML-tool-Alt-text-for-images--CHQ8mo21rEHWtEZOHiD6bFDlAg-UxE4cerPFJEp6OfqiGv0a

I have not copied anything verbatim from the Paper Doc so this text should be reviewed. I have not given alt text to the card images on the homepage or where they are reused in the app as they are decorative and surrounded by other text explaining the card/page.

### CODE REVIEW

File: src/messages/ui.en.json [hunk 1]
```diff
@@ -224,17 +226,23 @@
   "connectMB.USBCompatibility.transferStep.step1": "Open the location to which the firmware was downloaded. Most commonly found in your download folder.",
   "connectMB.USBCompatibility.transferStep.step2": "Drag the file onto the micro:bit on your computer's file explorer.",
   "connectMB.USBCompatibility.transferStep.step3": "Once the file has finished transferring, the micro:bit can be connected using Bluetooth.",
+  "connectMB.USBCompatibility.transferStep.altText": "animation of using drag and drop to program a micro:bit",
```
Comment (microbit-matt-hillsdon) [inline src/messages/ui.en.json:229]: Can this say something useful about what you drag where?

Comment (microbit-matt-hillsdon) [reply src/messages/ui.en.json:229]: Not an issue if that's in the surrounding copy.

**Comment by microbit-matt-hillsdon** [comment]
Confirmed all matches of `alt="[^"]` are in unused UI or brand/org names.

File: src/messages/ui.en.json (hunk 1 above)
Comment (microbit-robert) [reply src/messages/ui.en.json:229]: I think it's covered, and I'd argue if it's not clear from the point 2 in the dialog, then point 2 needs re-wording.

image

File: src/messages/ui.en.json [hunk 2]
```diff
@@ -156,12 +154,16 @@
...
+  "connectMB.connectCable.altText": "animation showing a USB cable being connected to the top of a micro:bit",
   "connectMB.connectCable.skip": "Skip: program already downloaded?",
 
   "connectMB.webPopup": "Select micro:bit",
   "connectMB.webPopup.instruction.heading": "In the next popup",
   "connectMB.webPopup.instruction1": "Choose your micro:bit",
-  "connectMB.webPopup.instruction2": "Select 'Connect'",
+  "connectMB.webPopup.webUsb.instruction2": "Select 'Connect'",
+  "connectMB.webPopup.webBluetooth.instruction2": "Select 'Pair'",
+  "connectMB.webPopup.webUsb.altText": "WebUSB connection dialog with BBC micro:bit selected",
```
Comment (microbit-matt-hillsdon) [inline src/messages/ui.en.json:165]: Python Editor uses:

WebUSB connection dialog with BBC micro:bit entry labelled 1 and Connect button labelled 2

Maybe we should match and do similarly for Bluetooth?

Comment (microbit-robert) [reply src/messages/ui.en.json:165]: I would have copied that exactly except that we haven't put the arrows / labels on/over the image yet.

**Review by microbit-matt-hillsdon — CHANGES_REQUESTED** [review]
A couple of comments otherwise LGTM

File: src/messages/ui.en.json (hunk 2 above)
Comment (microbit-matt-hillsdon) [reply src/messages/ui.en.json:165]: Ah yeah. Still though, I think we may as well just get the alt text right and then do the arrows when we get to them.

Comment (microbit-robert) [reply src/messages/ui.en.json]: I went for the optimistic approach that we'll do the labels [9123dc9](https://github.com/microbit-foundation/ml-trainer/pull/125/commits/9123dc94cdbac5acdd1ef801990698cfb972cbec).

File: src/messages/ui.en.json (hunk 1 above)
Comment (microbit-robert) [reply src/messages/ui.en.json]: Added more descriptive text [c6f12ec](https://github.com/microbit-foundation/ml-trainer/pull/125/commits/c6f12ecca81e6915bc4c4fcfd068427740abaa5e).

**Pull request merged**